sdk/eventhub/event-hubs/src/eventHubReceiver.ts

```diff
@@ -240,9 +255,20 @@ export class EventHubReceiver extends LinkEntity {
       }
     };
 
+    this._onAbort = () => {
+      const desc: string =
+        `[${this._context.connectionId}] The receive operation on the Receiver "${this.name}" with ` +
+        `address "${this.address}" has been cancelled by the user.`;
+      log.error(desc);
+      this._onError!(new Error(desc));
```

Comment:
This will only call the user's error handler, but the action of cancelling should cancel the entire receive operation. You can do that by calling `close()` which should also take care of removing the listener for the cancellation token

Reply:
I was also thinking the same but if we will call the `close()` inside the abort, then user can also call the `close()`,  if they want to cancel the request.
then what is the point of calling abort()?

Reply:
Use case difference:
- User can call stop() only if they have access to the receive handler returned by `receive` operation. Cancellation token allows cancelling from a different part of the user's application (where the token might have been created) which might not have access to the receive handler.

Behavior difference:
- Cancelling throws error, stop() doesnt

Reply:
>User can call stop() only if they have access to the receive handler returned by receive operation.

What about close()?  User can call close() receiver anytime.

Reply:
Yes, but then the expectation is that they have access to the receiver.